src/directLine.ts

```diff
@@ -869,4 +885,40 @@ export class DirectLine implements IBotConnection {
             )
         )
     }
+
+    private commonHeaders() {
+
+
+        if (this._userAgent === '') {
+            try {
+                this._userAgent = window.navigator.userAgent || '';
+            } catch {
+                try {
+                    // set node user agent
+                    // @ts-ignore
+                    const os = require('os');
+                    const { arch, platform, version } = process;
+                    this._userAgent = `Node.js,Version=${version}; ${platform} ${os.release()}; ${arch}`
+                } catch {
+                    // no-op
+                }
+            }
+        }
+
+        return {
```

Comment:
it would be nice to add some example agents in the PR description/commit message, for reference

Reply:
Meaning examples of passing in `options.botAgent`? Or a full example of the user-agent string?

Reply:
Some full examples of what the API will see

Reply:
Gotcha. I'll add that to the description.